monkeytype/stubs.py

```diff
@@ -503,22 +503,62 @@ def __repr__(self) -> str:
                                           tuple(self.function_stubs.values()),
                                           tuple(self.attribute_stubs))
 
-    @staticmethod
-    def stubs_from_typed_dict(type_dict: type, class_name: str) -> List['ClassStub']:
-        """Return a list of class stubs for all TypedDicts found within `type_dict`."""
-        assert is_anonymous_typed_dict(type_dict)
-        class_stubs = []
+
+class ReplaceTypedDictsWithStubs(TypeRewriter):
+    """Replace TypedDicts in a generic type with class stubs and store all the stubs."""
+
+    def __init__(self, class_name_hint: str) -> None:
+        self._class_name_hint = class_name_hint
+        self._stubs: List[ClassStub] = []
+
+    @property
+    def stubs(self) -> List[ClassStub]:
+        return self._stubs
+
+    def _rewrite_container(self, cls: type, container: type) -> type:
```

Comment:
Typically it's preferable to use `typing.Type` (with a subscript clarifying which types are accepted) rather than `type` for type annotations. But it looks like our `typing.pyi` stub forces your hand here, so that's probably an orthogonal change.

Reply:
Yeah, I'd have liked to use `Type[...]`. However, when I tried using just `cls: Type[List]` and left out the other possible container types, it typechecked even with `mypy --strict`. Not sure why it didn't catch the errors (probably because of the `type` annotations in `typing.pyi`).

I've left it as `type` as I'm not sure of the right type to use here.